source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -44,16 +44,30 @@ Http::FilterHeadersStatus Http1BridgeFilter::encodeHeaders(Http::ResponseHeaderM
     chargeStat(headers);
   }
 
-  if (!do_bridging_ || end_stream) {
+  if (!do_bridging_) {
+    return Http::FilterHeadersStatus::Continue;
+  }
+
+  response_headers_ = &headers;
+  if (end_stream) {
+    // We still need to check if would-be trailers are present as headers
+    // for a gRPC Trailers-Only response. Non-zero grpc-status will change the response code.
+    //
+    // However, we do not need to set grpc status and message in this case because the "trailers" we
+    // are passing here are already the response headers heading downstream to the client.
+    doTrailers(headers, /*set_grpc_status_and_message=*/false);
     return Http::FilterHeadersStatus::Continue;
   } else {
-    response_headers_ = &headers;
     return Http::FilterHeadersStatus::StopIteration;
   }
 }
 
 Http::FilterDataStatus Http1BridgeFilter::encodeData(Buffer::Instance&, bool end_stream) {
-  if (!do_bridging_ || end_stream) {
+  if (!do_bridging_) {
```

Comment:
one very small non-blocking nit: why change this?

Reply:
it felt right to continue the pattern from encodeHeaders above

Reply:
fair enough. i figured the change has no impact when encoding data.